src/app/client/src/app/modules/observation/components/observation-listing/observation-listing.component.ts

```diff
@@ -1,42 +1,328 @@
+    this.redoLayout();
+    this.layoutService
+      .switchableLayout()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe((layoutConfig) => {
+        if (layoutConfig != null) {
+          this.layoutConfiguration = layoutConfig.layout;
+        }
+        this.redoLayout();
+      });
+  }
+  redoLayout() {
+    if (this.layoutConfiguration != null) {
+      this.FIRST_PANEL_LAYOUT = this.layoutService.redoLayoutCSS(
+        0,
+        this.layoutConfiguration,
+        COLUMN_TYPE.threeToNine,
+        true
+      );
+      this.SECOND_PANEL_LAYOUT = this.layoutService.redoLayoutCSS(
+        1,
+        this.layoutConfiguration,
+        COLUMN_TYPE.threeToNine,
+        true
+      );
+    } else {
+      this.FIRST_PANEL_LAYOUT = this.layoutService.redoLayoutCSS(
+        0,
+        null,
+        COLUMN_TYPE.fullLayout
+      );
+      this.SECOND_PANEL_LAYOUT = this.layoutService.redoLayoutCSS(
+        1,
+        null,
+        COLUMN_TYPE.fullLayout
+      );
+    }
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
+
+  ngAfterViewInit() {
+    setTimeout(() => {
+      this.setTelemetryData();
+    });
+  }
+
+  private setTelemetryData() {
+    this.inViewLogs = []; // set to empty every time filter or page changes
+    this.telemetryImpression = {
+      context: {
+        env: this.activatedRoute.snapshot.data.telemetry.env,
+      },
+      edata: {
+        type: this.activatedRoute.snapshot.data.telemetry.type,
+        pageid: this.activatedRoute.snapshot.data.telemetry.pageid,
+        uri: this.userService.slug
+          ? "/" + this.userService.slug + this.router.url
+          : this.router.url,
+        subtype: this.activatedRoute.snapshot.data.telemetry.subtype,
+        duration: this.navigationhelperService.getPageLoadTime(),
+      },
+    };
+    this.cardIntractEdata = {
+      id: "content-card",
+      type: "click",
+      pageid: this.activatedRoute.snapshot.data.telemetry.pageid,
     };
+  }
 
-    this.kendraService.get(paramOptions).subscribe(data => {
-      debugger
-      console.log("========================== observation lis ====================");
-      console.log(data)
-    }, error => {
-      debugger
-    })
+  playContent(event) {
   }
 
+  public inView(event) {
+    _.forEach(event.inview, (elem, key) => {
+      const obj = _.find(this.inViewLogs, {
+        objid: elem.data.metaData.identifier,
+      });
+      if (!obj) {
+        this.inViewLogs.push({
+          objid: elem.data.metaData.identifier,
+          objtype: elem.data.metaData.contentType || "content",
+          index: elem.id,
+        });
+      }
+    });
+    if (this.telemetryImpression) {
+      this.telemetryImpression.edata.visits = this.inViewLogs;
+      this.telemetryImpression.edata.subtype = "pageexit";
+      this.telemetryImpression = Object.assign({}, this.telemetryImpression);
+    }
+  }
+
+  hoverActionClicked(event) {
```

Comment:
are we using this function?

Reply:
yes this is needed because it check the meta data identifier for card rendering

Reply:
hoveraction is used for card